modules/xml/filterx-func-format-xml.c

```diff
@@ -176,12 +177,65 @@ _append_entry(FilterXObject *key, FilterXObject *value, gpointer user_data)
   return TRUE;
 }
 
+static gboolean
+_is_valid_xml_tag_name(FilterXObject *tag, gpointer user_data)
+{
+  const gchar *tag_str;
+  gsize tag_str_len;
+
+  if (!filterx_object_extract_string_ref(tag, &tag_str, &tag_str_len))
+    {
+      FilterXFunctionFormatXML *self = ((gpointer *) user_data)[0];
+      gchar type_name_buf[FILTERX_OBJECT_TYPE_NAME_BUF_SIZE];
+      gchar *info = g_strdup_printf("Dict key must be a string, got %s",
+                                    filterx_object_format_type_name(tag, type_name_buf));
+      filterx_eval_push_error_info(XML_ERROR_STR, &self->super.super, info, TRUE);
+      return FALSE;
+    }
+
+  if (tag_str_len > 2 &&
+      (tolower(tag_str[0]) == 'x' &&
+       tolower(tag_str[1]) == 'm' &&
+       tolower(tag_str[2]) == 'l'))
+    {
+      FilterXFunctionFormatXML *self = ((gpointer *) user_data)[0];
+      filterx_eval_push_error_info(XML_ERROR_STR, &self->super.super,
+                                   "Dict key can't start with with \"xml\" string", FALSE);
+      return FALSE;
+    }
```

Comment:
If I understand the specs right, in an element's name, the prefix `xml` is valid, but it is reserved for standardized future use. So there can be a valid XML that has an element name starting with `xml`, as long as it follows the standardized reserved values (which currently there is not any, I believe, but it can be in the future).